contracts/asa_faucet/program.py

```diff
@@ -0,0 +1,109 @@
+
+from akita_inu_asa_utils import *
+from pyteal import *
+
+from pyteal import *
+import sys
+
+def approval_program():
+    # Keys for the global data stored by this smart contract.
+
+    # AssetID that this smart contract will drip.
+    asset_id_key = Bytes("asset_id")
+    # Time in seconds to allow drip
+    drip_time = Bytes("drip_time")
+    drip_amount = Bytes("drip_amount")
+
+    min_algo_amount = Bytes("min_algo_amount")
+    min_asset_amount = Bytes("min_asset_amount")
+
+    # keys for local data stored by this smart contract
+    user_last_claim_time = Bytes("last_claim_time")
+
+    @Subroutine(TealType.none)
+    def on_get_drip():
+        asset_holding = And(AssetHolding.balance(Txn.sender(), App.globalGet(asset_id_key)),
+                            Balance(Txn.sender()) >= min_algo_amount)
+        assert_fees = Assert(And(Global.group_size() == Int(2), Gtxn[0].amount() == Int(1000)))
+        assert_time = Assert(Global.latest_timestamp() - user_last_claim_time >= drip_time)
+        assert_has_volume = Assert(AssetHolding.balance(Global.current_application_address(), asset_id_key))

```

Comment:
AssetHolding.balance() returns a MaybeValue, which has no TEAL type and needs to be tested via the hasValue() and value() methods. Also needs to be sequenced before use for PyTeal to store in scratch ([reference](https://pyteal.readthedocs.io/en/latest/assets.html#balances)).

Most of the references to locals/globals here are to the bytestring keys directly, need to include global/local get calls.